client/build_mac_host.sh

```diff
@@ -1,25 +1,83 @@
 # Automates build process for Platform codegen on macOS.
 
-set -e
+set -ex
+
+# Defaults
+BUILD_TYPE="debug"
+
+# Parse CLI options
+while getopts ":ailrw" options; do
+    case $options in
+        a) FLAG_ANDROID=true ;;
+        i) FLAG_IOS=true ;;
+        l) FLAG_LINUX=true ;;
+        r) BUILD_TYPE="release" ;;
+        w) WORKAROUND_ROSETTA_ISSUE=true ;;
+    esac
+done
+
+## BUILD FOR CURRENT HOST ##
 
 CLIENT_DIR=$(realpath $(dirname $0))
 PLATFORM_DIR=$(realpath $CLIENT_DIR/..)
```

Comment:
```suggestion
CLIENT_DIR=$(readlink -f $(dirname $0))
PLATFORM_DIR=$(readlink -f $CLIENT_DIR/..)
```

This is more cross platform